docs/common/craft-parts/reference/plugins/dotnet_v2_plugin.rst

```diff
@@ -0,0 +1,231 @@
+.. _craft_parts_dotnet_v2_plugin:
+
+.NET plugin (v2)
+================
+
+The ``dotnet`` plugin builds .NET projects using the ``dotnet`` tool.
+
+.. note::
+    This plugin supersedes :ref:`.NET plugin <craft_parts_dotnet_plugin>`.
+
+Keywords
+--------
+
+In addition to the common :ref:`plugin <part-properties-plugin>` and
+:ref:`sources <part-properties-sources>` keywords, this plugin provides the following
+plugin-specific keywords:
+
+.. _global_flags:
+
+Global Flags
+~~~~~~~~~~~~
+
+dotnet-configuration
+^^^^^^^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``"Release"``
+
+The .NET build configuration to use. Possible values are ``"Debug"`` and ``"Release"``.
+
+dotnet-project
+^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``None``
+
+Path to the solution or project file to build relative to the root of the snap source.
+If a project or solution file is not specified, MSBuild will search t he root of the snap
+source for a file that has an extension that ends in either ``proj`` or ``sln``.
+
+dotnet-properties
+^^^^^^^^^^^^^^^^^
+**Type:** dict of strings to strings
+
+**Default:** Empty
```

Comment:
For configurations, "Unset" is typically the best word to indicate that the setting isn't used. It avoids the confusion around null/nil/noop.